src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
+                            href="/"
+                            sx={{...style.typography, ...style.typography.small}}
+                        >
+                            LOGO
+                        </Typography>
+                        
+                        <SmallView>
+                        <Box>                                
+                            {toggleButton}
+                        </Box>
+                        <IconButton
+                            sx={style.margin.iconBtn}
+                            aria-label="open drawer"
+                            edge="start"
+                            onClick={handleDrawerToggle}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                        </SmallView>
+                        <Responsive>
+                            <Box>
+                                <Button>
+                                    <Link style={Links} to="/posts/general">
+                                        General
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/learn">
+                                        Learn
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/interview">
+                                        Interview
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/posts/project">
+                                        Project
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/login">
+                                        SignIn
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/register">
+                                        SignUp
+                                    </Link>
+                                </Button>
+                                <Button>
+                                    <Link style={Links} to="/my-account/1">
+                                        Profile
+                                    </Link>
+                                </Button>
+                                {toggleButton}
+                            </Box>
+                        </Responsive>
+                    </Toolbar>
+                </Container>
+            </AppBar>
+            <SmallView>
+            <Box component="nav">
+                <Drawer
```

Comment:
We could consider using a SwipeableDrawer so touchscreens can swipe to open and close. Though this isn't strictly necessary